### PR TITLE
SKFP-354 Migration from persona to users-api (for Kidsfirst, will be removed after persona cut)

### PR DESCRIPTION
**Pull request by celinepelletier**

### CODE REVIEW

**Review by atoulous — APPROVED** [review]

File: src/routes/admin.ts [hunk 1]
```diff
@@ -25,4 +28,27 @@ adminRouter.put('/resetAllConsents', async (req, res, next) => {
     }
 });
 
+adminRouter.post('/doMigrationFromPersona', async (req, res, next) => {
+    try {
+        if (keycloakRealm !== Realm.KF) {
+            res.status(StatusCodes.BAD_REQUEST).send('Not available for this project');
```
Comment (evans-g-crsj) [inline src/routes/admin.ts:34]: a return statement would remove your else clause

File: src/utils/regex.test.ts
Comment (evans-g-crsj) [inline src/utils/regex.test.ts:1]: :+1:

File: src/db/models/User.ts [hunk 2]
```diff
@@ -234,6 +238,44 @@ UserModel.init(
...
+            allowNull: true,
+            validate: {
+                isAlpha: true,
+            },
+        },
+        location_country: {
+            type: DataTypes.CITEXT,
+            allowNull: true,
+            validate: {
+                validate: (value: string) => value === '' || NAME_REGEX.test(value),
```
Comment (evans-g-crsj) [inline src/db/models/User.ts:256]: should we trim the value?

File: src/external/persona.ts [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+
+    if (response.status === 200) {
+        return body;
+    }
+
+    throw new Error(`Error during call to Persona with status [${response.status}]`);
+};
+
+export const convertToPersona = (p: any): Persona => ({
+    id: p.id,
```
Comment (evans-g-crsj) [inline src/external/persona.ts:102]: if you need more complex computation you could use iife vs nested ternaries

File: src/external/persona.ts [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+    acceptedKfOptIn: boolean;
+    acceptedDatasetSub: boolean;
+    interests: string[];
+    egoId: string;
+    story: string;
+    bio: string;
+};
+
+export const validateUniqPersonas = (personas: Persona[]): boolean => {
+    const result = new Map();
```
Comment (evans-g-crsj) [inline src/external/persona.ts:49]: why not simply use an object?
const result = personnas.reduce((xs, x) => ..., {})

**Review by evans-g-crsj — APPROVED** [review]

**Review by adipaul1981 — APPROVED** [review]

**Review by evans-g-crsj — APPROVED** [review]

**Pull request merged**